### DISCUSSION TITLE
Quartz.NET 3.3.0

### DISCUSSION THREAD

**Discussion by lahma** [quartznet/quartznet#1154] [Announcements]
This release addresses problems with using Quartz with .NET Full Framework lower than 4.7.2. ValueTask loading
could fail due the dependencies brought with activity source support. Now activity sources are only supported when
using .NET Framework >= 4.7.2 and netstandard >= 2.0. This also raises requirement the same way for package
Quartz.OpenTelemetry.Instrumentation.

This release also improves trigger acquisition performance when using persistent job store, mostly by reducing network round-trips.
The semaphore implementations were also re-written to gain more performance.

Also some bug fixes included, thanks to all contributors!

## GitHub Issues

__BREAKING CHANGES__

* Activity source listener is not longer part of `net461` build, only `net472`
* `Quartz.AspNetCore` integration package minimum .NET Core version is now 3.1 for HealthChecks support
  
__NEW FEATURES__

* Separate build configuration for .NET Framework 4.7.2
* OpenTelemetry integration upgraded to target OpenTelemetry 1.0.0-rc1.1
* Ported `JobInterruptMonitorPlugin` from Java version which allows automatic interrupt calls for registered jobs (#1110)
* Rewrite semaphore implementations (#1115)
* `UsingJobData` now has `Guid` and `char` overloads (#1141)
* Add a regular `AddJob(Type)` (#1090)
  
__FIXES__

* Jobs not firing after upgrade to 3.2.x (from 3.0.7) on Microsoft Server 2008 R2 (#1083)
* Jobs are not fired (#1072)
* `MicrosoftDependencyInjectionJobFactory` does not inject job properties for scoped jobs (#1106)
* XSD schema no longer requires defining `durable` element if you just want to define `recover` (#1128)
* Stack trace logging fixed in case of reporting invalid lock acquire (#1133)
* Disposable job is disposed twice when using `UseMicrosoftDependencyInjectionScopedJobFactory` (#1120)
* `QuartzHostedService.StopAsync` throws `NullReferenceException` if `StartAsync` hasn't been run (#1123)

<Download />

<hr /><em>This discussion was created from the release <a href='https://github.com/quartznet/quartznet/releases/tag/v3.3.0'>Quartz.NET 3.3.0</a>.</em>

**Comment by ebonato** [comment]
Please, re-publish package with strong name, as in previous package

![image](https://user-images.githubusercontent.com/2773529/113903400-52d5ad80-97a7-11eb-837f-2677cc9c0fc7.png)

**Reply by lahma** [reply]
My Spanish (?) is a bit poor, but the DLLs should be signed, checked with NuGet Package Explorer:

![image](https://user-images.githubusercontent.com/171892/113904251-8ed63600-97da-11eb-83fc-12366b5a17c5.png)

Can this be some other dependency or something similar?

**Reply by ebonato** [reply]
Yep. It's a dependency... 
Nuget package states that dependencies over Logging.Abstractions and DiagnosticSource could be equal or higher than 2.1.1 and 4.7.1, 
but if you use higher versions of these packages, it fails to load. 
JustDecompile is showing that Quartz.dll is linked with specific version of these dependencies.

![image](https://user-images.githubusercontent.com/2773529/113919479-09428e00-97ba-11eb-89bc-12b4564560bb.png)

![image](https://user-images.githubusercontent.com/2773529/113913288-87029b80-97b2-11eb-81ee-5f0d3feb7eee.png)

**Reply by lahma** [reply]
Higher versions should be allowed. Do you have explicit binding redirects?

**Reply by ebonato** [reply]
Yes. upgrading quartz packages via nuget, also update binding redirects on web.config...
I'm trying to figure out the root cause using a new sample web applications.. 
Tks for your support.

**Reply by ebonato** [reply]
Just creating a new web project (Full Framework @ 4.7.2) with visual studio, and adding new Quartz packages via nuget, throws assembly load error (due strong name). 

There is any code at web application startup to raise an Scheduller instance.
Using previous package version, works as expected.

Its something related to internals of assembly signing. It's failing at assembly load.

[Quatz.WebApplicationFull472.zip](https://github.com/quartznet/quartznet/files/6274241/Quatz.WebApplicationFull472.zip)

**Reply by erinloy** [reply]
+1

**Reply by lahma** [reply]
@ebonato @erinloy Quartz.NET 3.3.1 has been released addressing this signing problem, sorry for the trouble.

**Reply by ebonato** [reply]
Tks a lot @lahma..
First time hearing about [public signing](https://github.com/dotnet/runtime/blob/main/docs/project/public-signing.md).
I must add that using an ASP.NET Core Web App, version 3.3.0 was working as expected.... loading assembly with public signing...
Public signing should be great for Quartz (or any other community package), as will help to improve tailored versions of the package, without breaking dependency chain...
